lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/driver/GroupCommDriverImpl.java

```diff
@@ -124,6 +126,19 @@ public GroupCommDriverImpl(final ConfigurationSerializer confSerializer,
                              @Parameter(DriverIdentifier.class) final String driverId,
                              @Parameter(TreeTopologyFanOut.class) final int fanOut,
                              final LocalAddressProvider localAddressProvider) {
+    this(confSerializer, driverId, fanOut, localAddressProvider, new MessagingTransportFactory());
+  }
+
+  /**
+   * @deprecated Have an instance injected instead.
+   */
+  @Deprecated
```

Comment:
We want to have an injected instance with Tang, instead of calling a constructor directly.